examples/rifle_sxiv.sh

```diff
@@ -10,39 +10,55 @@
 #
 #   mime ^image, has sxiv, X, flag f = path/to/this/script -- "$@"
 #
-# Implementation notes: this script is quite slow because of POSIX limitations
-# and portability concerns. First calling the shell function 'abspath' is
-# quicker than calling 'realpath' because it would fork a whole process, which
-# is slow. Second, we need to append a file list to sxiv, which can only be done
-# properly in two ways: arrays (which are not POSIX) or \0 sperated
-# strings. Unfortunately, assigning \0 to a variable is not POSIX either (will
-# not work in dash and others), so we cannot store the result of listfiles to a
+# Implementation note: the script tries to be POSIX compliant both in terms of
+# shell syntax and calls being made to external utilities, such as grep or find.
+# This makes it portable across many unix like systems, although it may not be
+# the cleanest or fastest approach.
+#
+# First, using case statement to get absolute path is quicker than calling
+# 'realpath' because it would fork a whole process, which is slow.
+#
+# Second, we need to append a file list to sxiv, which can only be done
+# properly in three ways: arrays (which are not POSIX) or \0 sperated strings.
+# Unfortunately, assigning \0 to a variable is not POSIX either (will not work
+# in dash and others), so we cannot store the result of listfiles to a
 # variable.
+#
+# The third approach is to store the result to a tmpfile and using `-i` to feed
+# the list to sxiv. This is the fastest approach since we won't have to call
+# listfiles twice.
 
-if [ $# -eq 0 ]; then
-    echo "Usage: ${0##*/} PICTURES"
-    exit
-fi
 
-[ "$1" = '--' ] && shift
+tmp="/tmp/sxiv_rifle_$$"
```

Comment:
Maybe we should use `mktemp`? I'm not sure whether `tee` might append if we do happen to get a repeated pid. And it'd also avoid hard-coding `/tmp`.

Reply:
Oh, and the "Three ways, '1' or '2'. '2' isn't posix. The third is..." reads a bit confusing tbh.

Reply:
I don't believe `mktemp` is posix.
Is repeated pid possible?
What should be done then if we want to avoid hardcoding `/tmp`

Reply:
Repeated pids are unlikely but possible, yes.

I think this is a case where support is widespread enough that it's still in the spirit of POSIX. Maybe using `TMPDIR` instead of `/tmp` would be enough though.

Reply:
On my system at least `$TMPDIR` doesn't exist. Falling back to `/tmp` should be saner I guess.

Reply:
Cleaned up the comment to be more natural https://github.com/ranger/ranger/pull/2411/commits/72a0cfefcfb349627f0d278cb1bc295f0a242bbc.

Should we deal with repeated pid? One way I can think of is running a loop checking if `$tmp` exists and keep appending more stuff to `$tmp` until we get something that doesn't already exist. Do you have any cleaner suggestion?

Reply:
Nah, unlikely anyway. And if `tee` overwrites rather than appends it's a non-issue anyway.

Reply:
Then we should be good here.

Reply:
We should still use `${TMPDIR:-/tmp}` or something though.

Reply:
I've done that already.
https://github.com/N-R-K/ranger/pull/2411/commits/4451a87a2bf9b95afd566ddcaaa7ba548d1ec8dc